### PR TITLE
Create a database table for user blocks

### PR DESCRIPTION
**Pull request by hellcp-work**
Introduces a database table for associating users with whoever they block. We use this for creating associations between users. This way a user can block multiple users.

### CODE REVIEW

File: src/api/app/models/user_block.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class UserBlock < ApplicationRecord
```
Comment (krauselukas) [inline src/api/app/models/user_block.rb:1]: @hellcp-work I'm a bit unsure about the naming... We start having more of these places where something is blocked. Maybe `UserBlockedUser` would be a bit more precise?

File: src/api/app/models/user_block.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  belongs_to :user, optional: false
+  belongs_to :blocked, class_name: 'User', optional: false
+
+  validates :blocked_id, uniqueness: { scope: :user_id, message: 'This user is already blocked.' }
+end
+
+# == Schema Information
+#
+# Table name: user_blocks
```
Comment (krauselukas) [inline src/api/app/models/user_block.rb:10]: then the table would be called `blocked_users`

**Comment by danidoni** [comment]
Would you mind explaining a bit why this is needed in the description?

File: src/api/app/models/blocked_user.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+# == Schema Information
+#
+# Table name: blocked_users
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  blocked_id :integer          not null, indexed, indexed => [user_id]
+#  user_id    :integer          not null, indexed => [blocked_id]
```
Comment (danidoni) [inline src/api/app/models/blocked_user.rb:16]: I would rename this to `blocker_id`

File: src/api/app/models/user_block.rb (hunk 1 above)
Comment (danidoni) [reply src/api/app/models/user_block.rb]: Fair enough

**Review by saraycp — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**